### DISCUSSION TITLE
Retry GitLab API requests on HTTP 409 Resource lock errors

### DISCUSSION THREAD

**Discussion by fgreinacher** [renovatebot/renovate#28972] [Suggest an Idea]
### Tell us more.

The GitLab API might return `HTTP 409 Resource lock` in cases where an internal DB lock fails to be acquired in time, see https://gitlab.com/gitlab-org/gitlab/-/blob/443c12cf3b238385db728f03b2cdbb4f17c70292/lib/api/api.rb#L111. Those errors are usually transient and a retry is therefore worthwhile.

The `python-gitlab` library for example adding special handling for this in https://github.com/python-gitlab/python-gitlab/pull/2326.

What do you think about adding similar logic to Renovate's GitLab layer?

**Comment by rarkins** [comment]
Sounds good

**Reply by HonkingGoose** [reply]
See issue:

- #29006
